xlators/storage/posix/src/posix-inode-fd-ops.c

```diff
@@ -4225,7 +4228,8 @@ posix_fgetxattr(call_frame_t *frame, xlator_t *this, fd_t *fd, const char *name,
         if (size == 0)
             goto done;
     }
-    list = alloca(size + 1);
+
+    list = GF_MALLOC(size, gf_posix_mt_char);
```

Comment:
previously it was size + 1, now its size, is that OK? (Also everywhere else, its only 'size').

Reply:
Better to keep it size+1. In case it exposes any bugs in old clients it will be a problem. Since we are using GF_MALLOC, better to set the last byte to zero as well.

Reply:
I believe it is not required, everywhere we use in similar fashion like posix_getxattr also.If you think still it can create an issue i will update in next patch.For specific to last byte llistxattr itself return NULL terminated character string so i feel it is no required.

Reply:
Each name in the list is null terminated. The list itself may not be null terminated, but the code doesn't rely on that, so I would say it's fine (`alloca()` didn't clear the memory either).